File: android/ui-models/src/main/kotlin/com/gemwallet/android/ui/models/name/NameRecordState.kt
package com.gemwallet.android.ui.models.name

import com.wallet.core.primitives.NameRecord

sealed interface NameRecordState {
    data object None : NameRecordState
    data object Loading : NameRecordState
    data object Error : NameRecordState
    data class Complete(val record: NameRecord) : NameRecordState

    val nameRecord: NameRecord?
        get() = (this as? Complete)?.record
}

File: ios/Packages/PrimitivesComponents/Sources/Types/NameRecordState.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Foundation
import Primitives

public enum NameRecordState: Equatable, Hashable, Sendable {
    case none
    case loading(name: String)
    case error
    case complete(NameRecord)
}

public extension NameRecordState {
    var result: NameRecord? {
        switch self {
        case let .complete(result): result
        case .none, .loading, .error: .none
        }
    }

    var requestedName: String? {
        switch self {
        case let .loading(name): name
        case let .complete(record): record.name
        case .none, .error: .none
        }
    }
}
